go/nbs/dynamo_manifest.go

```diff
@@ -39,13 +40,14 @@ type ddbsvc interface {
 type dynamoManifest struct {
 	table, db string
 	ddbsvc    ddbsvc
+	lock      addr
```

Comment:
Yeah, I don't have a problem with this implementation, but abstractly, AWS share:

1-table file format
2-table naming
3-root hash encoding (20 byte hash).

And differ on 
1-Manifest encoding

The one difference makes sense because we need to optimistically lock on an attribute in Dynamo.

I think it makes sense that both implementations should be required to compute the same logical lock hash given the same root hash and included table hashes.